ksql-engine/src/main/java/io/confluent/ksql/engine/QueryEngine.java

```diff
@@ -42,7 +42,7 @@
 import org.slf4j.LoggerFactory;
 
 // CHECKSTYLE_RULES.OFF: ClassDataAbstractionCoupling
-class QueryEngine {
+public class QueryEngine {
```

Comment:
I don't think we should make this public.  It's an internal impl detail of the engine and should IMHO remain that way.

If you just parse, prepare + execute the query against a sandbox of the engine, (as the rest api normally does), then you'll get back a query, and you can get the schema from that query.

Yes, this is executing some bits we don't need. But it maintains better decoupling in the code IMHO.

If we _really_ want to expose logical plans, then we should expose them _through_ the engine.

Reply:
I'm happy with your suggestion and will do that.